pkg/services/hcloud/server/server.go

```diff
@@ -465,8 +465,30 @@ func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
 				s.scope.Name(),
 				err,
 			)
-			return nil, errServerCreateNotPossible
+
+			err = errors.Join(errServerCreateNotPossible, err)
```

Comment:
ah now I saw what this error is used for. I think that in both cases that we have there, we should return this error as well to not immediately reconcile. WDYT?

Reply:
seems logical ... then it would reque after 5 min...

Reply:
so it this correctly implemented right now?

Reply:
I guess it is. then we can resolve it

Reply:
Yes